### PR TITLE
Support customize redpanda service name

### PR DESCRIPTION
**Pull request by sniperking1234**
https://github.com/redpanda-data/helm-charts/issues/644

### CODE REVIEW

**Review by joejulian — APPROVED** [review]
Just needs a minor version bump of the Chart.yaml

**Comment by sniperking1234** [comment]
@joejulian done.

File: charts/redpanda/templates/_helpers.tpl [hunk 1]
```diff
@@ -33,6 +33,17 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+{{/*
+Create a default service name
+*/}}
+{{- define "redpanda.servicename"}}
+{{- if .Values.service.name}}
+{{- .Values.service.name | trunc 63 | trimSuffix "-" -}}
+{{- else }}
+{{ include "redpanda.fullname" . | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end}}
```
Comment (joejulian) [inline charts/redpanda/templates/_helpers.tpl:45]:
```suggestion
{{- define "redpanda.servicename" -}}
{{- if dig "service" "name" false .Values -}}
{{- .Values.service.name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{ include "redpanda.fullname" . | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
```
If we use dig here, we won't need to require that the service key exists.

Also, added trailing `-` to match our style for named templates.

File: charts/redpanda/values.yaml [hunk 2]
```diff
@@ -56,6 +56,10 @@ image:
   # If `image.tag` is 'latest', the default is `Always`.
   pullPolicy: IfNotPresent
 
+# -- Redpanda Service settings.
+service:
+  name: redpanda
+
```
Comment (joejulian) [inline charts/redpanda/values.yaml:62]:
```suggestion
# -- Redpanda Service settings.
# service:
#   -- set service.name to override the default service name
#   name: redpanda

```

**Review by joejulian — CHANGES_REQUESTED** [review]
CI is failing because you included a default service.name that changes the default behavior.

On upgrades, this fails because the existing external service that was named for the release and the new one named redpanda have conflicting ports.

I've added a couple suggestions that should solve that.

File: charts/redpanda/templates/_helpers.tpl (hunk 1 above)
Comment (sniperking1234) [reply charts/redpanda/templates/_helpers.tpl]: Thank you for your suggestion.

**Comment by joejulian** [comment]
Sorry about the bad suggestion. I pushed a fix.

One last thing, could you squash this PR please?

**Comment by sniperking1234** [comment]
@joejulian It's already done.

**Review by joejulian — APPROVED** [review]
LGTM!

**Pull request merged**